e2e-tests/playwright/support/pages/CatalogImport.ts

```diff
@@ -56,47 +54,47 @@ export class BackstageShowcase {
     await this.page.click(BackstageShowcasePO.tablePreviousPage);
   }
 
-  async clickLastPage(e) {
+  async clickLastPage() {
     await this.page.click(BackstageShowcasePO.tableLastPage);
   }
 
   async clickFirstPage() {
     await this.page.click(BackstageShowcasePO.tableFirstPage);
   }
-
-  async verifyPRRowsPerPage(rows: number, allPRs: any[]) {
+  async verifyPRRowsPerPage(rows, allPRs) {
     await this.selectRowsPerPage(rows);
-    const rowsCount = await this.page
-      .locator(BackstageShowcasePO.tableRows)
-      .count();
-    if (rowsCount !== rows) {
-      throw new Error(`Expected row count to be ${rows}, but got ${rowsCount}`);
-    }
-    await this.page.locator(`text=${allPRs[rows - 1].title}`).isVisible();
-    const numberExists = await this.page
-      .locator(`text=#${allPRs[rows].number}`)
-      .isVisible();
-    if (numberExists) {
-      throw new Error(
-        `Element with number ${allPRs[rows].number} should not exist`,
-      );
-    }
+    const rowElement = this.page.locator(`text=${allPRs[rows - 1].title}`);
+    await rowElement.scrollIntoViewIfNeeded();
+    await expect(rowElement).toBeVisible();
+
+    const nonExistentElement = this.page.locator(
+      `a:text("#${allPRs[rows].number}")`,
+    );
+    await expect(nonExistentElement).not.toBeVisible();
+
+    const tableRows = this.page.locator(BackstageShowcasePO.tableRows);
+    await expect(tableRows).toHaveCount(rows);
   }
 
   async selectRowsPerPage(rows: number) {
     await this.page.click(BackstageShowcasePO.tablePageSelectBox);
     await this.page.click(`ul[role="listbox"] li[data-value="${rows}"]`);
   }
 
+  async getWorkflowRuns() {
+    const response = await APIHelper.githubRequest(
+      'GET',
+      githubAPIEndpoints.workflowRuns,
+    );
+    const responseBody = await response.json();
+    return responseBody.workflow_runs;
+  }
+
   async verifyPRStatisticsRendered() {
     const regex = /Average Size Of PR\d+ lines/;
-    const statsVisible = await this.page
-      .locator('tr')
-      .filter({ hasText: regex })
-      .isVisible();
-    if (!statsVisible) {
-      throw new Error('PR statistics are not rendered');
-    }
+    const stats = this.page.locator('tr').filter({ hasText: regex });
```

Comment:
I think you can use uiHelper for all those generic situations, and move `waitFor` to the method inside UIHelper